<launch>

  <arg name="stereo" default="stereo_forward"/>
  <arg name="image" default="image_rect_color"/>

  <node pkg="valve_tracker_triton" type="tracker_node" name="valve_tracker" output="screen">
    <param name="stereo_frame_id" value="/stereo_forward_optical"/>
    <param name="mean_filter_size" value="5"/>
    <param name="opening_element_size" value="10"/>
    <param name="closing_element_size" value="14"/>
    <param name="epipolar_width_threshold" value="4"/>
    <param name="binary_threshold" value="250"/>
    <param name="min_value_threshold" value="39"/>
    <param name="min_blob_size" value="5"/>
    <param name="max_blob_size" value="200"/>
    <param name="show_debug" value="true"/>
    <param name="tf_filter_size" value="3"/>  

    <rosparam command="load" file="$(find valve_tracker_triton)/etc/model.yaml" />

    <remap from="stereo" to="$(arg stereo)"/>
    <remap from="image" to="$(arg image)"/>

  </node>

  <!-- Static transform from valve to connector -->
    <node pkg="tf" type="static_transform_publisher" name="valve_to_connector" args="0 0.39 -0.15 0 0 0.52 valve connector 100" />
  
</launch>
